node/network/statement-distribution/src/vstaging/tests/grid.rs

```diff
@@ -14,28 +14,2981 @@
+						FullStatementWithPVD::Seconded(c, p)
+							if c == &candidate && p == &pvd
+					);
+				}
+			);
+		}
+
+		// Receive Backed message.
+		overseer
+			.send(FromOrchestra::Communication {
+				msg: StatementDistributionMessage::Backed(candidate_hash),
+			})
+			.await;
+
+		// Should send an acknowledgement back to C.
+		{
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages:: NetworkBridgeTx(
+					NetworkBridgeTxMessage::SendValidationMessage(
+						peers,
+						Versioned::VStaging(
+							protocol_vstaging::ValidationProtocol::StatementDistribution(
+								protocol_vstaging::StatementDistributionMessage::BackedCandidateKnown(ack),
+							),
+						),
+					)
+				) => {
+					assert_eq!(peers, vec![peer_c]);
+					assert_eq!(ack, BackedCandidateAcknowledgement {
+						candidate_hash,
+						statement_knowledge: StatementFilter {
+							seconded_in_group: bitvec::bitvec![u8, Lsb0; 0, 1, 1],
+							validated_in_group: bitvec::bitvec![u8, Lsb0; 0, 0, 0],
+						},
+					});
+				}
+			);
+
+			answer_expected_hypothetical_depth_request(&mut overseer, vec![], None, false).await;
+		}
+
+		// TODO: Receive a manifest about the same candidate from peer D. Contains different statements.
+		{
+			let manifest = BackedCandidateManifest {
+				relay_parent,
+				candidate_hash,
+				group_index: other_group,
+				para_id: other_para,
+				parent_head_data_hash: pvd.parent_head.hash(),
+				statement_knowledge: StatementFilter {
+					seconded_in_group: bitvec::bitvec![u8, Lsb0; 1, 1, 0],
+					validated_in_group: bitvec::bitvec![u8, Lsb0; 0, 0, 0],
+				},
+			};
+
+			send_peer_message(
+				&mut overseer,
+				peer_d.clone(),
+				protocol_vstaging::StatementDistributionMessage::BackedCandidateManifest(
+					manifest.clone(),
+				),
+			)
+			.await;
+
+			let expected_ack = BackedCandidateAcknowledgement {
+				candidate_hash,
+				statement_knowledge: StatementFilter {
+					seconded_in_group: bitvec::bitvec![u8, Lsb0; 0, 1, 1],
+					validated_in_group: bitvec::bitvec![u8, Lsb0; 0, 0, 0],
+				},
+			};
+
+			// Instantaneously acknowledge.
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages:: NetworkBridgeTx(
+					NetworkBridgeTxMessage::SendValidationMessages(messages)
+				) => {
+					assert_eq!(messages.len(), 2);
+					assert_eq!(messages[0].0, vec![peer_d]);
+					assert_eq!(messages[1].0, vec![peer_d]);
+
+					assert_matches!(
+						&messages[0].1,
+						Versioned::VStaging(protocol_vstaging::ValidationProtocol::StatementDistribution(
+							protocol_vstaging::StatementDistributionMessage::BackedCandidateKnown(ack)
+						)) if *ack == expected_ack
+					);
+
+					assert_matches!(
+						&messages[1].1,
+						Versioned::VStaging(protocol_vstaging::ValidationProtocol::StatementDistribution(
+							protocol_vstaging::StatementDistributionMessage::Statement(r, s)
+						)) if *r == relay_parent && s.unchecked_payload() == &CompactStatement::Seconded(candidate_hash)
```

Comment:
```suggestion
						)) if *r == relay_parent && s.unchecked_payload() == &CompactStatement::Seconded(candidate_hash) && s.unchecked_validator_index() == v_e
```

Reply:
I suspect this check should pass and also answers the question here https://github.com/paritytech/polkadot/pull/6755/files/d0477d6e5668924a80ab0d2bc6b8e165a3428fde#r1123615222